scout/adapter/mongo/clinvar.py

```diff
@@ -393,8 +393,11 @@ def delete_clinvar_object(self, object_id: str, object_type: str, submission_id:
             self.clinvar_collection.delete_one({"_id": object_id})
 
         # in any case remove reference to it in the submission object 'case_data' list field
+
         self.clinvar_submission_collection.find_one_and_update(
-            {"_id": ObjectId(submission_id)}, {"$pull": {"case_data": object_id}}
+            {"_id": ObjectId(submission_id)},
+            {"$pull": {"case_data": {"$regex": f"^{re.escape(object_id)}"}}},
+            return_document=ReturnDocument.AFTER,
```

Comment:
case_data ID starts with variant_data ID, but contains also sample name:

<img width="946" alt="image" src="https://github.com/user-attachments/assets/c7c65f20-e7ac-4445-831b-2a7181d8eb06" />

Reply:
I had a Quick Look at the code. It appears the variable `object_id` rather contains either a `case_id` or `subm_variant_id`, both of which start with the `case_id` str, and neither of which is an `ObjectId`, based on the contents of the `delete_object` buttons. This should be fine. But the naming is a little unfortunate. 😆 Sounds good with a joined document in only one collection later! 👍